### ISSUE TITLE
It seems terminating an instance should detach non-root volumes too

### ISSUE THREAD

**Issue by tbarrella** [spulec/moto#1922]
With moto 1.3.6, if I do something like
```
import boto3
import moto

@moto.mock_ec2
def test_detach_volumes():
    region = 'us-west-1'
    ec2 = boto3.resource('ec2', region_name=region)
    volume = ec2.create_volume(AvailabilityZone=region, Size=10)

    instance, = ec2.create_instances(ImageId='ami-12345678', MinCount=1, MaxCount=1)
    volume.attach_to_instance(Device='/dev/sdb1', InstanceId=instance.id)
    instance.terminate()
    volume.load()
```
`volume.attachments` will say that it's still attached to the instance, whereas I find in production that terminating an instance will detach non-root volumes in addition to deleting the root volume. Right now `Instance.teardown_defaults()` includes deleting the root volume, but it seems it should also detach other volumes, and potentially delete those with "DeleteOnTermination" set

**Comment by spulec** [comment]
Thanks; setting as a feature request.

**Comment by bblommers** [comment]
This is available as of moto >= 1.3.15.dev871
